File: src/main/java/com/github/xumeng/java/problems/easy/problem0263/Solution0263.java
package com.github.xumeng.java.problems.easy.problem0263;

public class Solution0263 {

    public boolean isUgly(int n) {
        if (n <= 0) {
            return false;
        }
        int quotient = n;
        for (int i = 1; i < quotient; i++) {
            quotient = n / i;
            if (n % i != 0) {
                continue;
            }
            if (isPrime(i) && i > 5) {
                return false;
            }
            if (isPrime(quotient) && quotient > 5) {
                return false;
            }
        }
        return true;
    }

    private boolean isPrime(int n) {
        for (int i = 2; i < n / i; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0263/Solution0263.kt
package com.github.xumeng.kotlin.problems.easy.problem0263

class Solution0263 {

    fun isUgly(n: Int): Boolean {
        if (n <= 0) {
            return false
        }
        var quotient: Int = n
        var i = 1
        while (i <= quotient) {
            quotient = n / i
            if (n % i != 0) {
                i++
                continue
            }
            if (isPrime(i) && i > 5) {
                return false
            }
            if (isPrime(quotient) && quotient > 5) {
                return false
            }
            i++
        }
        return true
    }

    private fun isPrime(n: Int): Boolean {
        var i = 2
        while (i <= n / i) {
            if (n % i == 0) {
                return false
            }
            i++
        }
        return true
    }

}